x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/ip/IpPrefix.java

```diff
@@ -136,7 +136,14 @@ static BytesRef process(
             throw new IllegalArgumentException("Prefix length v6 must be in range [0, 128], found " + prefixLengthV6);
         }
 
-        boolean isIpv4 = Arrays.compareUnsigned(ip.bytes, 0, IPV4_PREFIX.length, IPV4_PREFIX, 0, IPV4_PREFIX.length) == 0;
+        boolean isIpv4 = Arrays.compareUnsigned(
+            ip.bytes,
```

Comment:
Oh good catch. Lots of  `BytesRef`s do indeed start at 0, but the ones we use might not.